packages/task/src/browser/task-configurations.ts

```diff
@@ -0,0 +1,257 @@
+    }
+
+    setClient(client: TaskConfigurationClient) {
+        this.client = client;
+    }
+
+    dispose() {
+        this.toDispose.dispose();
+    }
+
+    watchWorkspaceConfiguration(wsRootUri: string): Promise<void> {
+        return this.watchConfigurationFile(wsRootUri);
+    }
+
+    protected watchConfigurationFile(wsRootUri: string): Promise<void> {
+        return new Promise(resolve => {
+            let configFile: string;
+            this.getConfigFile(wsRootUri)
+                .then(uri => {
+                    // start watching that config file
+                    configFile = uri;
+                    if (this.watchedConfigFileUris.indexOf(configFile) === -1) {
+                        this.watchedConfigFileUris.push(configFile);
+                        this.watchConfigFile(configFile);
+                        return;
+                    }
+
+                })
+                .then(() => {
+                    this.readConfig(configFile)
+                        .then(tasks => {
+                            for (const task of tasks) {
+                                this.tasksMap.set(task.label, task);
+                            }
+                            resolve();
+                        })
+                        .catch(err => {
+                            this.logger.error(`Workspace: ${err}`);
+                        });
+                })
+                .catch(err => {
+                    this.logger.error(`Workspace: ${err}`);
+                });
+        });
+    }
+
+    protected readConfig(uri: string): Promise<TaskOptions[]> {
+        return new Promise((resolve, reject) => {
+            const tasksFound: TaskOptions[] = [];
+
+            // read tasks defined in tasks.json
+            this.readTasks(uri)
+                .then(tasks => {
+                    for (const task of tasks) {
+                        tasksFound.push(task);
+                    }
+                    resolve(tasksFound);
+                })
+                .catch(err => {
+                    reject(err);
+                });
+        });
+    }
+
+    protected getConfigFile(rootDir: string): Promise<string> {
+        return new Promise((resolve, reject) => {
+            // compute URL of the tasks file to watch
+            const cfgFile = new URI(rootDir).resolve(this.TASKFILEPATH).resolve(this.TASKFILE).toString();
+            this.fileSystem.exists(cfgFile).then(exists => {
+                if (exists) {
+                    resolve(cfgFile);
+                } else {
+                    this.logger.error(`config file tasks.json does not exist under ${rootDir}`);
+                    reject(`config file does not exist under: ${rootDir}`);
+                }
+            });
+        });
+    }
+
+    getTaskLabels(): string[] {
+        return [...this.tasksMap.keys()];
+    }
+
+    getTask(taskLabel: string): Promise<TaskOptions> {
+        return new Promise((resolve, reject) => {
```

Comment:
why is this async?

Reply:
No good reason that I can see - I made this synchronous instead.

Reply:
no need for async - made sync